lib/event_grid.jl

```diff
@@ -0,0 +1,75 @@
+
+###########################################
+## Interpolated event grid (requires Grid)
+###########################################
+
+@comment """
+# Interpolated event grid
+"""
+
+using Requires
+
+@require Grid begin
+
+    using Grid
+    
+    function search_grid (g::InterpGrid, x::Float64)
+        i = searchsortedfirst (g, x)
+        g[i]
+    end
+
+    function grid_point (x::Float64, v::Float64)
+        x - v
+    end
+    
+@doc* """
+Interpolated event grid (requires Grid library).
+
+```julia
+
+EventGrid(events, tf, dt)
+```
+
+Given a vector with event times and a sampling step, creates a
+continuous interpolated function that has a negative value between
+events, and crosses zero at each event time.
+The function `grid_input` can be used to detect events in model equations.
+
+### Arguments
+
+* `event` : Vector of event times
+* `tf` : grid end time
+* `dt` : event grid sampling time
+
+### Example
+
+```julia
+
+input = EventGrid([1.0,2.0,3.0],tstop,0.1)
```

Comment:
The sampling time is just the interval used to create the interpolation
grid. E.g. if you are reading events from an external source, then the
sampling time would need to be 1/2F where F is the average sample
frequency. If there is a better way to do this, I am very open to
suggestions.

On Sun, Feb 8, 2015 at 3:35 PM, Tom Short <notifications@github.com> wrote:

> In lib/event_grid.jl
> <https://github.com/tshort/Sims.jl/pull/52#discussion_r24304022>:
>
> > +Given a vector with event times and a sampling step, creates a
> > +continuous interpolated function that has a negative value between
> > +events, and crosses zero at each event time.
> > +The function `grid_input` can be used to detect events in model equations.
> > +
> > +### Arguments
> > +
> > +* `event` : Vector of event times
> > +* `tf` : grid end time
> > +* `dt` : event grid sampling time
> > +
> > +### Example
> > +
> > +```julia
> > +
> > +input = EventGrid([1.0,2.0,3.0],tstop,0.1)
>
> I'm probably being dense. I assume there's an event every 1 sec in this
> example. What happens every 0.1 sec (the sampling time)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tshort/Sims.jl/pull/52/files#r24304022>.
>